### PR TITLE
WIP: Add support for Course Expiration and Feature Based Enrollment

### PR DESCRIPTION
**Pull request by stvstnfrd**
This depends on https://github.com/edx/edx-platform/pull/23670

### CODE REVIEW

File: src/courseware/course/sequence/Sequence.jsx [hunk 1]
```diff
@@ -131,6 +135,9 @@ function Sequence({
               handlePrevious();
             }}
           />
+          { courseExpiredMessage && (
```
Comment (davidjoy) [inline src/courseware/course/sequence/Sequence.jsx]:
```
<Alert type="info">
  <div dangerouslySetInnerHTML={{ __html: courseExpiredMessage }} />
</Alert>
```

File: src/courseware/course/sequence/Unit.jsx [hunk 2]
```diff
@@ -40,6 +49,27 @@ export default function Unit({
...
       />
+      { contentTypeGatingEnabled && unit.graded && verifiedMode && enrollmentMode === 'audit' && (
+        <div className="unit-content-container content-paywall">
+          <div>
+            <h4>
+              <FontAwesomeIcon icon={faLock} />
+              <span>Verified Track Access</span>
+            </h4>
+            <p>
+              Graded assessments are available to Verified Track learners.
```
Comment (davidjoy) [inline src/courseware/course/sequence/Unit.jsx]: i18n

Comment (davidjoy) [reply src/courseware/course/sequence/Unit.jsx]: FormattedMessage
injectIntl
intl

File: src/access-expiration-alert/AccessExpirationAlert.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+
+import Alert from '../user-messages/Alert';
+
+function AccessExpirationAlert({ intl }) {
+  return (
+    <Alert type="info">
+      <div dangerouslySetInnerHTML={{ __html: courseExpiredMessage }} />
```
Comment (stvstnfrd) [inline src/access-expiration-alert/AccessExpirationAlert.jsx]: @davidjoy I've converted my custom alert to use the AlertList framework, but the last bit I'm missing is how do I share state/context (`course.courseExpiredMessage` or `expiredMessage`) from `hooks.js`?

Comment (stvstnfrd) [reply src/access-expiration-alert/AccessExpirationAlert.jsx]: I checked the other `Alert` classes and didn't see them doing anything similar;
their messages don't rely on any of the data used/calculated in their respective `hooks.js`.

**Comment by stvstnfrd** [comment]
Closing in favor of:
- https://github.com/edx/frontend-app-learning/pull/49
- https://github.com/edx/frontend-app-learning/pull/50

**Pull request closed without merging**